### DISCUSSION TITLE
Any events on websocket-client disconnect?

### DISCUSSION THREAD

**Discussion by a-marchenko** [nats-io/nats-server#2175] [Q&A] [answered]
I need to know that the client (nats.ws) has disconnected without constantly polling all clients every few seconds.
I'm wondering if there are some NATS events that I could subscribe to in other services?
Or how do I need to modify the nats-server code in order to timely learn about the broken connection to the websocket client?

**Comment by aricart** [comment, marked as answer]
Are you listening for disconnect events?

https://github.com/nats-io/nats.ws#lifecycleinformational-events

**Comment by nuharaf** [comment]
You ask for NATS event, so probably this one is what you are looking for 
[https://docs.nats.io/nats-server/configuration/sys_accounts#available-events-and-services](https://docs.nats.io/nats-server/configuration/sys_accounts#available-events-and-services)
